api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3209,5 +3220,165 @@ public void saveRetrospectiveOrder_shouldFailIfAnActiveDrugOrderForTheSameConcep
 		expectedException.expect(APIException.class);
 		expectedException.expectMessage("Cannot have more than one active order for the same orderable and care setting");
 		orderService.saveRetrospectiveOrder(order, null);
+    }
+	
+	@Test
+	public void shouldSaveOrdersWithSortWeightWhenWithinAOrderGroup() throws Exception {
+		executeDataSet(ORDER_SET);
+		
+		Encounter encounter = encounterService.getEncounter(3);
+		
+		OrderSet orderSet = Context.getOrderSetService().getOrderSetById(2000);
+		OrderGroup orderGroup = new OrderGroup();
+		orderGroup.setOrderSet(orderSet);
+		
+		orderGroup.setEncounter(encounter);
+		
+		Order newOrder1 = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1000)
+				.withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+				.withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+		newOrder1.setOrderGroup(orderGroup);
+		
+		Order newOrder2 = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1001)
+				.withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+				.withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+		newOrder2.setOrderGroup(orderGroup);
+		
+		Order newOrder = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1000)
+				.withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+				.withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date()).build();
+		
+		//Saving three orders, two with an orderGroup and one without an orderGroup 
+		Context.getOrderService().saveOrderGroups(Arrays.asList(newOrder1, newOrder2, newOrder));
+		Context.getOrderService().saveOrder(newOrder, null);
+		Context.flushSession();
+		
+		OrderGroup savedOrderGroup = Context.getOrderService().getOrderGroupByUuid(orderGroup.getUuid());
+		Order savedOrder = Context.getOrderService().getOrderByUuid(newOrder.getUuid());
+		
+		assertEquals("The first order in the orderGroup gets sortWeight 10.0",new Double(10.0), savedOrderGroup.getOrders().get(0).getSortWeight());
+		assertEquals("The first order in  savedOrderGroup is the same which is sent first in the List", newOrder1.getUuid(), savedOrderGroup.getOrders().get(0).getUuid());
+		
+		assertEquals("The second order in the orderGroup gets sortWeight 20.0",new Double(20.0), savedOrderGroup.getOrders().get(1).getSortWeight());
+		assertEquals("The second order in  savedOrderGroup is the same which is sent second in the List",newOrder2.getUuid(), savedOrderGroup.getOrders().get(1).getUuid());
+		assertNull("The order which doesn't belong to an orderGroup has no sortWeight", savedOrder.getSortWeight());
```

Comment:
The numbers 10 and 20 are not a function of OrderService. We should probably just assert that the sequencing is correct. (order1.getSortWeight() < order2.getSortWeight()).